### ISSUE TITLE
🔗 ♿ 🐞 Bug Report: How to disable the URL in the title accordion Elementor? (Merged with #14371) - [ED-8824]

### ISSUE THREAD

**Issue by Bambi-O** [elementor/elementor#8943]
Hey,

I have a problem with accordion of elementor. Do you know if it’s possible to disable the integration automatic of URL in the title of the accordion ? 

Thanks in advance 

Eloi

<img width="927" alt="Capture d’écran 2562-08-30 à 09 56 34" src="https://user-images.githubusercontent.com/54695948/63990545-90c01980-cb0e-11e9-8dab-5b1e619f78e4.png">

**Comment by shilo-ey** [comment]
Hi @Bambi-O 

The Accordion widget does not change the URL, but showing you the URL at the bottom status bar due to the fact that each title is a link that toggles between open and closed state.

Try adding this jQuery code to replace the `a` tags to `span`
```JS
jQuery( document ).ready(function( $ ){
    $(".elementor-tab-title a").replaceWith(function() { 
        return "<span>" + this.innerHTML + "</span>"; 
    });
});
```

**Comment by ravikiran-srini** [comment]
Hi @shilo-ey

Can you please tell me where exactly do I add that jquery code?

**Comment by StiefTschobbs** [comment]
Hi @shilo-ey, 

it would be really great if you can give us a hint, where to add the jquery code. I've tried to change (a --> span) and deleted the "href" directly in the toogle.php, but there was no effect on it. Could you please support us?

Thanks a lot in advance.

**Comment by devslayertech** [comment]
Please fix this in the next version of Elementor:

![image](https://user-images.githubusercontent.com/100967476/166464604-a63eb5c9-06c5-4589-9d79-a2fb7105937f.png)

**Comment by yllip** [comment]
+1

**Comment by Marc-DATA-HORIZON** [comment]
Hi Team Elementor, for me the Case is still Open as my Website delays all JS files for the Best Page Score. But now as Elementor used a A-Tag for Title without href="#" and ask to add a jQuery script to solve this Issue? I think that is not correct, also, as we wait over 3 years for that easy solution.

**Comment by luvstuss** [comment]
Agree, the problem didn't solve. Seems that Elementor developers did as it was comfortable for them, but didn't follow the Google guidelines.

**Comment by AndyMik90** [comment]
> Hi @Bambi-O
> 
> The Accordion widget does not change the URL, but showing you the URL at the bottom status bar due to the fact that each title is a link that toggles between open and closed state.
> 
> Try adding this jQuery code to replace the `a` tags to `span`
> 
> ```js
> jQuery( document ).ready(function( $ ){
>     $(".elementor-tab-title a").replaceWith(function() { 
>         return "<span>" + this.innerHTML + "</span>"; 
>     });
> });
> ```

I've included this code on my site, and the accordions appear as span, but I still get an empty ahref tag in my source code and it gives errors in Google Page speed insights.
"<a href="" class="elementor-toggle-title">"

**Comment by Sebastian-Habiger** [comment]
Hi,
I think PHP fits better to solve this issue. Here is an elementor hook that does the same.
```PHP
add_filter( 'elementor/widget/render_content', function ( $widget_content, $widget ) {
    if ( 'accordion' === $widget->get_name() ) {
        $widget_content = preg_replace(
            '/<a class="elementor-accordion-title" href="">(.*?)<\/a>/',
            '<span class="elementor-accordion-title">$1</span>',
            $widget_content
        );
    }

    return $widget_content;
}, 10, 2 );
```

**Comment by AndyMik90** [comment]
> Hi, I think PHP fits better to solve this issue. Here is an elementor hook that does the same.
> 
> ```
> add_filter( 'elementor/widget/render_content', function ( $widget_content, $widget ) {
>     if ( 'accordion' === $widget->get_name() ) {
>         $widget_content = preg_replace(
>             '/<a class="elementor-accordion-title" href="">(.*?)<\/a>/',
>             '<span class="elementor-accordion-title">$1</span>',
>             $widget_content
>         );
>     }
> 
>     return $widget_content;
> }, 10, 2 );
> ```

Doesn't seem to do the trick on my page [https://mikalsenutvikling.no/](https://mikalsenutvikling.no/) still getting a href class on elementor toggle title.

**Comment by Sebastian-Habiger** [comment]
> > Hi, I think PHP fits better to solve this issue. Here is an elementor hook that does the same.
> > ```
> > add_filter( 'elementor/widget/render_content', function ( $widget_content, $widget ) {
> >     if ( 'accordion' === $widget->get_name() ) {
> >         $widget_content = preg_replace(
> >             '/<a class="elementor-accordion-title" href="">(.*?)<\/a>/',
> >             '<span class="elementor-accordion-title">$1</span>',
> >             $widget_content
> >         );
> >     }
> > 
> >     return $widget_content;
> > }, 10, 2 );
> > ```
> 
> Doesn't seem to do the trick on my page https://mikalsenutvikling.no/ still getting a href class on elementor toggle title.
Actually on your page the a tag looks like: ```<a href="" class="elementor-toggle-title">```
Thereforew the RegEx does not fit and you need to adapt. Something like this.
```PHP
add_filter( 'elementor/widget/render_content', function ( $widget_content, $widget ) {
    if ( 'accordion' === $widget->get_name() ) {
        $widget_content = preg_replace(
            '/<a[^>]*class="elementor-accordion-title"[^>]*>(.*?)<\/a>/',
            '<span class="elementor-accordion-title">$1</span>',
            $widget_content
        );
    }

    return $widget_content;
}, 10, 2 );
```

**Comment by AndyMik90** [comment]
> 

Thanks for your reply, but still does not work. Is it because that the class is "elemenetor-toggle-title" and not "elementor-accordion-title"?

**Comment by Sebastian-Habiger** [comment]
Sorry, yes, my bad, did not see that.

**Comment by AndyMik90** [comment]
For reference to others, I got this code to work if the accordion version does not:

`add_filter( 'elementor/widget/render_content', function ( $widget_content, $widget ) {
    if ( 'toggle' === $widget->get_name() ) {
        $widget_content = preg_replace(
            '/<a[^>]*class="elementor-toggle-title"[^>]*>(.*?)<\/a>/',
            '<span class="elementor-toggle-title">$1</span>',
            $widget_content
        );
    }

    return $widget_content;
}, 10, 2 );`

**Comment by armanism24** [comment]
Hi @AndyMik90 ,
Can you tell me where to put this code on my Wordpress site? i am also getting this issue on google pagespeed insights 
Thanks

**Comment by AndyMik90** [comment]
> Hi @AndyMik90 , Can you tell me where to put this code on my Wordpress site? i am also getting this issue on google pagespeed insights Thanks

Put the code in your function.php file

**Comment by armanism24** [comment]
seems to be working, thanks 😊

**Comment by gorkemuluser** [comment]
Hello, I added it to the functions.php file but unfortunately it didn't work. The theme I use is called JupiterX. Could it be related to this or can we try with a different method?

**Comment by gorkemuluser** [comment]
By the way, it looks like I found a solution.

1. I opened the elementor/includes/widgets/accordion.php file.
2. I removed the **href=""** in `<a class="elementor-accordion-title" href=""> `on line 578.

It's working for now. But I'm not sure what will happen after the update.

**Comment by WebRat45** [comment]
I fixed this by adding a # to href="" on line 578 of accordion.php. Lighthouse no longer mentions the links are not crawlable.

**Comment by avazargozalli** [comment]
I found a very simple solution to this problem. _(I am using this issue for toggle element. You can use the same solution for accordion.)_ 
Here's the solution (Do not use the # sign so that the page does not return to the top.):

**1. Open:** elementor/includes/widgets/toggle.php
**2. Find:** 
`<a href="" class="elementor-toggle-title"><?php $this->print_unescaped_setting( 'tab_title', 'tabs', $index ); ?></a>`
change like this;
`<a href="#toggle" class="elementor-toggle-title"><?php $this->print_unescaped_setting( 'tab_title', 'tabs', $index ); ?></a>`
And Save
**3. Edit the Toggle element**  (On the page where you will use the element)
Advanced > CSS ID > Add 'toggle' text
And Save

This will not only fix your Lighthouse Score, but will also prevent your page from returning to the top after clicking, and will always keep it in the Toggle (accordion) element.

**Comment by bethgee** [comment]
The above is a trick but will reset every time you update the plugin, no?? Is there really no long term solution here?

**Comment by avazargozalli** [comment]
Unfortunately, there is no permanent solution for now.

**Comment by Sebastian-Habiger** [comment]
What about the content filter solution mentioned a few comments earlier? 
They are pretty robust and the regexp can be tweaked if something changes.

https://github.com/elementor/elementor/issues/8943#issuecomment-1340855278
https://github.com/elementor/elementor/issues/8943#issuecomment-1340703967
https://github.com/elementor/elementor/issues/8943#issuecomment-1340855278

**Comment by bethgee** [comment]
I tried all of those, Sebastian, with no difference!


> What about the content filter solution mentioned a few comments earlier? They are pretty robust and the regexp can be tweaked if something changes.
> 
> [#8943 (comment)](https://github.com/elementor/elementor/issues/8943#issuecomment-1340855278) [#8943 (comment)](https://github.com/elementor/elementor/issues/8943#issuecomment-1340703967) [#8943 (comment)](https://github.com/elementor/elementor/issues/8943#issuecomment-1340855278)

**Comment by Sebastian-Habiger** [comment]
It seems like the Elementor team is messing with the order of attributes so the regexp would not work. We need to adapt that in a more generic way to address that too. I could try to give a better regexp that although will work for both toggle and accordion.

**Comment by bethgee** [comment]
The client is really keen to get it sorted, so if you get the chance that would be amazing!! I don't want a solution that requires messing with the code every update...

Any chance you know how we can also fix another issue I just posted about ??  - https://github.com/elementor/elementor/issues/22021

**Comment by WebRat45** [comment]
Any permanent fixes?

**Comment by bethgee** [comment]
Nope, nothing yet

**Comment by gorkemuluser** [comment]
First, go to your WordPress dashboard.

Navigate to the "Custom code" section in the Elementor plugin.
Add the following code snippet:

`<script>
jQuery(document).ready(function() {
    jQuery('.elementor-accordion-title').removeAttr('href');
});
</script>
`

Save the changes.

**Comment by bethgee** [comment]
> First, go to your WordPress dashboard.
> 
> Navigate to the "Custom code" section in the Elementor plugin. Add the following code snippet:
> 
> `<script> jQuery(document).ready(function() { jQuery('.elementor-accordion-title').removeAttr('href'); }); </script> `
> 
> Save the changes.

This works - thank you!!!

Any chance you know how to fix the aria role issue I am also having - [#22021](https://github.com/elementor/elementor/issues/22021) ?

**Comment by avazargozalli** [comment]
This code also works for Toggle. I did not encounter any warnings in the Google site speed test.

```
<script> jQuery(document).ready(function() { jQuery('.elementor-accordion-title').removeAttr('href'); }); </script>
<script> jQuery(document).ready(function() { jQuery('.elementor-toggle-title').removeAttr('href'); }); </script>
```

**Comment by noelawill** [comment]
thanks guys. Will this work for tabs too?  If so is the correct selector?

.elementor-tabs-title

**Comment by nicholaszein** [comment]
Hello, everyone!

Thank you for your feedback! 🙏 

### 🔗 🔃 🐞 This issue is being merged with #14371.

Which is on the same subject. 

For this reason, I'm closing this thread so we can have all the discussions in one location, and deal with it more efficiently.

Feel free to continue the conversation there. 😉 

Kind regards

**Comment by nicholaszein** [comment]
Hello!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor v3.14.0! 🥳 

Now Accordions are completely a11y compatible! 💪 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
